lib/scope.js

```diff
@@ -179,6 +179,30 @@ module.exports = {
     return safer_name;
   },  // Ends scope.getSafeScenarioFilename()
 
+  getJSONFilename: async function( scope ) {
+    let filename = await scope.getSafeScenarioFilename( scope, {
+      prefix: `downloads_${ scope.page_id }_JSON`
+    });
+    filename += `.json`;
+    return filename;
+  },
```

Comment:
From my JS ignorance: why does this need to be `await`ed? Or maybe the question should be why is `getSafeScenarioFilename` async? Nothing in it seems to be dependent on IO, it's just string concatenation.

Reply:
You're right about async - because it's async it needs `await`. It's async because everything's in scope right now and I made scope consistent. When I call something in scope, I wasn't always sure what needed to be `await`ed and what didn't, so I made it all async. There's an issue out there somewhere about refactoring name/string manipulation into its own file, and other such refactoring, so that not everything is in scope and not everything needs to be async.